### ISSUE TITLE
How to use classes in a module which contains a class of the same name (SideMenuController)

### ISSUE THREAD

**Issue by seasox** [mac-cain13/R.swift#407]
I use [SideMenuController](https://github.com/teodorpatras/SideMenuController) in my app. The `SideMenuController` module contains two custom segues, `CenterContainmentSegue` and `SideContainmentSegue`, but also a `SideMenuController` class.
During build, the following code is generated and build fails with `SideContainmentSegue is not a member type of SideMenuController`:

```
/// Segue identifier `ShowStatistic`.
static let showStatistic: Rswift.StoryboardSegueIdentifier<SideMenuController.CenterContainmentSegue, App.SideMenuControllerSplitter, UIKit.UINavigationController> = Rswift.StoryboardSegueIdentifier(identifier: "ShowStatistic")
/// Segue identifier `SideMenu`.
static let sideMenu: Rswift.StoryboardSegueIdentifier<SideMenuController.SideContainmentSegue, App.SideMenuControllerSplitter, MenuController> = Rswift.StoryboardSegueIdentifier(identifier: "SideMenu")
```
At first, I tried removing the explicit module in my Storyboard, causing a runtime error (`Could not create a segue of class '(null)'`). I was able to workaround this problem by introducing typealiases for these custom segues:

```
typealias _CenterContainmentSegue = CenterContainmentSegue
typealias _SideContainmentSegue = SideContainmentSegue

extension SideMenuController {
    typealias CenterContainmentSegue = _CenterContainmentSegue
    typealias SideContainmentSegue = _SideContainmentSegue
}
```

Maybe there is a more elegant way to fix this in R.swift?

Thank you for your work on R.swift, it helps a lot getting rid of stringly typed code.

**Comment by mac-cain13** [comment]
A class with the same name as the module it is in always gives trouble. You are hitting this Swift compiler issue: https://bugs.swift.org/browse/SR-898

Could you try the workaround [described here](https://github.com/mac-cain13/R.swift/issues/277#issuecomment-266575695):

- Remove the explicit import by emptying the module field in the xib/storyboard for the problematic module
- Make sure the module is imported by adding the new --import SomeModule argument to the R.swift build phase

**Comment by seasox** [comment]
Thanks for the response @mac-cain13 
I did exactly that, which caused the runtime error above (`Could not create a segue of class '(null)'`). The issue in my case is that there is a class named `SideMenuController`, but also classes named `SideContainmentSegue` and `CenterContainmentSegue`, which are *not* nested classes of `SideMenuController` but are in the module `SideMenuController`.

**Comment by mac-cain13** [comment]
I think a combination of using `--import SideMenuController` to make sure the import will be in the generated file and then _not_ stating an explicit module will generate the same code as in a normal use case.

If that doesn't work I'm curious what code R.swift generates for you and how you would write it by hand so that it does work. If we know the difference we could look into how to fix it.

**Comment by seasox** [comment]
Not stating an explicit module triggers the `NSInternalConsistencyException`: `SIGABRT` in performSegue,`UIViewController+StoryboardSegueIdentifierProtocol.swift`, Line 27. So I need to state the module explicitly, I guess. [I created a gist](
https://gist.github.com/seasox/b93444ec4162cfb302fb91d1c7cabe65) with a diff between the working and broken R.generated.swift files. I also added the workaround I wrote to make it work. Maybe something like the workaround could be generated automatically by `rswift`?
